src/app/components/translate-atttributes/translate-attributes.component.ts

```diff
@@ -0,0 +1,303 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Resource } from '../../models/resource';
+import { ResourceType } from '../../models/resource-type';
+import { System } from '../../models/system';
+import { AttributeTranslation } from '../../models/attribute-translation';
+import { ResourceService } from '../../service/resource/resource.service';
+import { SystemService } from '../../service/system.service';
+import { ResourceTypeService } from '../../service/resource-type.service';
+import { AttributeTranslationService } from '../../service/attribute-translation.service';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'admin-translate-attributes',
+  templateUrl: './translate-attributes.component.html',
+  styleUrls: ['./translate-attributes.component.css'],
+})
+export class TranslateAttributesComponent implements OnInit {
+  @Input() resource: Resource;
+  resourceTypes: ResourceType[];
+  systems: System[];
+  saving: Boolean = false;
+  errorMessage: string;
+  multipleActionsResults: {
+    text: String;
+    type: String;
+  }[] = [];
+
+  constructor(
+    protected resourceService: ResourceService,
+    protected systemService: SystemService,
+    protected resourceTypeService: ResourceTypeService,
+    protected activeModal: NgbActiveModal,
+    protected attributeTranslationService: AttributeTranslationService,
+  ) {}
+
+  ngOnInit(): void {
+    this.loadAttributes();
+  }
+
+  loadAttributes(): void {
+    this.attributeTranslationService
+      .getAttributes(this.resource.id)
+      .then((res) => {
+        this.resource = res;
+      });
+  }
```

Comment:
because we are loading the attributes in `ngOnInit()` and then setting the `resource` property with the result, maybe separate the `@Input` property to just be the `resource.id` from the `ResourceComponent`.

I'm seeing a bit of flicker of the old attributes when I load the modal after having saved several changes to the translated attributes.

Reply:
Added a resource Id input to prevent flicker and added text on the modal to show users if they're no current translated attributes.